### PR TITLE
Table: fix sort icon

### PR DESCRIPTION
**Pull request by bezany**
Reproduce link: https://codepen.io/anon/pen/BeygzE
After click `Sort by Name` sort icon from `Date` column not hidden.
I found that need clear `sortingColumn.order` when change sort column.

### CODE REVIEW

**Comment by ziyoung** [comment]
Please resolve conflict.

**Comment by bezany** [comment]
@ziyoung Done! May be more elegant added fix [here](https://github.com/ElemeFE/element/blob/f6bab114e3f8126a942938cf59d1afe5b09d9337/packages/table/src/store/watcher.js#L276), but I don't fully understand code.

**Comment by ziyoung** [comment]
> @ziyoung Done! May be more elegant added fix [here](https://github.com/ElemeFE/element/blob/f6bab114e3f8126a942938cf59d1afe5b09d9337/packages/table/src/store/watcher.js#L276), but I don't fully understand code.

What is your question?

File: packages/table/src/store/index.js [hunk 1]
```diff
@@ -70,6 +70,9 @@ Watcher.prototype.mutations = {
   sort(states, options) {
     const { prop, order } = options;
     if (prop) {
+      if (states.sortingColumn) {
```
Comment (ziyoung) [inline packages/table/src/store/index.js]: You can add these lines in `$nextTick` callback.

File: test/unit/specs/table.spec.js [hunk 2]
```diff
@@ -1758,6 +1758,52 @@ describe('Table', () => {
         });
       }, DELAY);
     });
+
+    it('sort correct change icon', done => {
```
Comment (ziyoung) [inline test/unit/specs/table.spec.js]: For asynchronous testing, use the async function.

File: test/unit/specs/table.spec.js [hunk 3]
```diff
@@ -1758,6 +1758,52 @@ describe('Table', () => {
...
+        template: `
+          <el-table ref="table" :data="testData" >
+            <el-table-column prop="name" sortable />
+            <el-table-column prop="release" sortable />
+            <el-table-column prop="director" sortable />
+            <el-table-column prop="runtime" sortable />
+          </el-table>
+        `,
+
+        created() {
```
Comment (ziyoung) [inline test/unit/specs/table.spec.js]: Why do you set `testData` in the created hook? You can set `testData` in data function.
```js
data() {
  return { testData: getTestData() }
}
```

**Comment by island205** [comment]
@bezany Please rebase `dev` branch and resolve conflict.

File: test/unit/specs/table.spec.js (hunk 3 above)
Comment (bezany) [reply test/unit/specs/table.spec.js]: You right, thanks.

File: test/unit/specs/table.spec.js (hunk 2 above)
Comment (bezany) [reply test/unit/specs/table.spec.js]: I found utils function `waitImmediate`. Good!

File: packages/table/src/store/index.js (hunk 1 above)
Comment (bezany) [reply packages/table/src/store/index.js]: I moved fix in another function.

**Comment by bezany** [comment]
@ziyoung Thank you for review! I improved code.
@island205 Did rebase and fix conflict.
Now fix in [`updateSort`](https://github.com/bezany/element/blob/fix/table_sort/packages/table/src/store/watcher.js#L269) function.

When I fix issue I saw, that code in [`handleSortClick` function](https://github.com/ElemeFE/element/blob/f6bab114e3f8126a942938cf59d1afe5b09d9337/packages/table/src/table-header.js#L473) have same code as in [`updateSort` function](https://github.com/ElemeFE/element/blob/f6bab114e3f8126a942938cf59d1afe5b09d9337/packages/table/src/store/watcher.js#L276). And property `sortingColumn.order` look as duplicate of `sortOrder`. May be planning this changes for future refactoring.

**Pull request merged**